src/coreclr/src/gc/gc.cpp

```diff
@@ -20636,9 +20636,19 @@ void gc_heap::mark_phase (int condemned_gen_number, BOOL mark_only_p)
     if (gc_t_join.joined())
 #endif //MULTIPLE_HEAPS
     {
+        uint64_t total_promoted_bytes = 0;
 #ifdef HEAP_ANALYZE
         heap_analyze_enabled = FALSE;
-        GCToEEInterface::AnalyzeSurvivorsFinished(condemned_gen_number);
+#ifdef MULTIPLE_HEAPS
+        for (int i = 0; i < n_heaps; i++)
+        {
+            promoted_bytes (i);
```

Comment:
>promoted_bytes (i); [](start = 12, length = 19)

you still need to add this to total_promoted_bytes though.

`total_promoted_bytes += promoted_bytes (i);`

also I would suggest to rename total_promoted_bytes to something else since there's already a PER_HEAP total_promoted_bytes. perhaps call it something like `promoted_bytes_global`.